feature/google/src/main/java/com/revenuecat/purchases/google/BillingFlowParamsExtensions.kt

```diff
@@ -2,12 +2,13 @@ package com.revenuecat.purchases.google
 
 import com.android.billingclient.api.BillingFlowParams
 import com.revenuecat.purchases.common.ReplaceProductInfo
+import com.revenuecat.purchases.models.GoogleProrationMode
 
 fun BillingFlowParams.Builder.setUpgradeInfo(replaceProductInfo: ReplaceProductInfo) {
     val subscriptionUpdateParams = BillingFlowParams.SubscriptionUpdateParams.newBuilder().apply {
         setOldPurchaseToken(replaceProductInfo.oldPurchase.purchaseToken)
         replaceProductInfo.prorationMode?.let {
-            setReplaceProrationMode(it)
+            setReplaceProrationMode((it as GoogleProrationMode).playBillingClientMode)
```

Comment:
This could potentially crash if for some reason we passed a non-GoogleProrationMode here and try to cast it... We could do something like:
```
(replaceProductInfo.prorationMode?.let {
   val googleProrationMode = it as? GoogleProrationMode
   if (googleProrationMode == null) {
      errorLog("Got non-Google proration mode")
   } else {
      setReplaceProrationMode(googleProrationMode.playBillingClientMode)
   }
}
```

Reply:
What's the issue with actually crashing? This is a programming error... and a pretty bad one. This would mean you're using the Google implementation and suddenly passing in Amazon objects, so I'd be fine with it crashing. IMO this would be even better than just logging an error that could go undetected.

Reply:
I kind of agree with you that crashes are much more visible and could help detect issues sooner. However, I believe the policy we've been following in the SDKs is that we really don't want to cause crashes in our SDK so we always program defensively and try to fail gracefully. This could be a discussion for the whole team though. CC @aboedo, In case you have any thoughts.

Reply:
Note that we are crashing devs apps here, and we're not sure in what situation their app might be, or how they would like to handle. That's the main reason we try to avoid crashing whenever possible.